ipv8/dht/community.py

```diff
@@ -458,6 +463,18 @@ def on_find_response(self, peer, dist, payload):
             cache.deferred.callback((cache.node, {'values': payload.values} if payload.values
                                      else {'nodes': payload.nodes}))
 
+    def block_chunk_refresh(self):
+        """
+        Republish any TrustChain block chunks which are about to expire in the DHT
+        """
+        from ..REST.dht_endpoint import TC_KEY_SUFFIX
+        hashed_dht_key = hashlib.sha1(self.my_peer.public_key.key_to_bin() + TC_KEY_SUFFIX).digest()
+
+        # for value in self.storage.get(hashed_dht_key):
+        for value in self.storage.items[hashed_dht_key]:
+            if value.age + TC_BLOCK_DELTA > value.max_age:
```

Comment:
We can't wait until we get (close to) max_age. Nodes on the network continuously join/leave the network, and chances are that all nodes that originally stored our values will have left the network. It's better to simply republish your latest block hourly (assuming you haven't created any new block for the past hour).